src/transforms/groupby.js

```diff
@@ -117,14 +117,14 @@ function transformOne(trace, state) {
     var opts = state.transform;
     var groups = trace.transforms[state.transformIndex].groups;
 
-    var groupNames = groups.filter(function(g, i, self) {
-        return self.indexOf(g) === i;
-    });
-
     if(!(Array.isArray(groups)) || groups.length === 0) {
         return trace;
     }
 
+    var groupNames = groups.filter(function(g, i, self) {
+        return self.indexOf(g) === i;
+    });
```

Comment:
done in https://github.com/plotly/plotly.js/pull/1020/commits/f6f072f09a769a5c3e050be11cde584c7ec475b0